src/components/concentration-display/Concentration.tsx

```diff
@@ -51,12 +56,43 @@ const Concentration: React.FC<AgentProps> = ({
                 />
             );
         } else {
+            let percentage;
+            const startingConcentration = concentration[agent];
+            if (startingConcentration !== undefined) {
+                percentage =
+                    (startingConcentration / maxConcentration) *
+                    widthMinusMargins;
+            }
```

Comment:
Replacing Ln 59 with `let percentage: number | undefined = undefined;` might better signal that percentage is in a valid state rather than an accidentally uninitialized one if the `if` never sets it.